### ISSUE TITLE
Problem with image overwrite

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#289]
**Reported by valleegr on 21 Dec 2006 04:47 UTC**
From Jeff Squyres:

If I build an oscar image and then go back, make some changes, and
then try to build it again, I get a popup box saying "Are you sure
you want to overwrite this image?" (paraphrase).  If I click "yes",
it fails to build the image.  I have to name my new image something
different than the original image.  Is there a reason for this?

**Comment by dikim33** [comment]
**Comment by valleegr on 22 Dec 2006 03:05 UTC**
Update: To reproduce the bug, follow these steps:
1/ build OSCAR clients image,
2/ define OSCAR clients (at least one),
3/ in a console, type 'si_lsimage', the image will appear
4/ go directly back to the step "build OSCAR clients image", and try to overwrite the image, it will fail
5/ in a console, type 'si_lsimage', the image is not any more there and image's files are still there (/var/lib/systemimager/images/...).

Note also that if you delete clients before to try to overwrite the image, it works. I do not know if it is really an OSCAR bug or a SIS bug.

**Comment by dikim33** [comment]
**Comment by bli on 29 Jan 2007 04:20 UTC**
This is an issue with SystemInstaller.

If the user choose to replace the image, the code would execute mksiimage and delete the image.  This will fail if clients are associated with that particular image, as can be seen from the log:

```
=============================================================================
== Running step 4 of the OSCAR wizard: Build OSCAR client image
=============================================================================

--> Step 4: Identified distro of clients: fc 6
--> Step 4: distro repo: http://download.fedora.redhat.com/pub/fedora/linux/core/6/i386/os/
--> Step 4: oscar repo: /tftpboot/oscar/common-rpms,/tftpboot/oscar/fc-6-i386
--> Step 4: Using RPM list: /root/dev/oscar/oscarsamples/fc-6-i386.rpmlist
Image oscarimage is assigned to clients.
 at /usr/bin/mksiimage line 300
Executing command: mksiimage -A --name oscarimage --location /tftpboot/oscar/common-rpms,/tftpboot/oscar/fc-6-i386,http://download.fedora.redhat.com/pub/fedora/linux/core/6/i386/os/ --filename /root/dev/oscar/oscarsamples/fc-6-i386.rpmlist --arch i686 --path /var/lib/systemimager/images/oscarimage --filename=/tmp/oscar-install-rpmlist.10009
Image oscarimage exists -- not added.
 at /usr/bin/mksiimage line 141
Image build finished.
Added Disk Table for oscarimage based on /root/dev/oscar/oscarsamples/ide.disk

FATAL: Image "oscarimage" doesn't appear to exist!

Couldn't run si_mkautoinstallscript -quiet --autodetect-disks -image oscarimage -force -ip-assignment static -post-install reboot at /usr/lib/perl5/vendor_perl/5.8.8/i386-linux-thread-multi/Tk.pm line 247
Image oscarimage is assigned to clients.
 at /usr/bin/mksiimage line 300
```

Will have to think about how to fix this.

**Comment by dikim33** [comment]
**Modified by bli on 29 Jan 2007 09:46 UTC**

**Comment by dikim33** [comment]
**Comment by bli on 4 Feb 2007 02:58 UTC**
(In [5663]) - Removed MANIFEST file, this is generated from MANIFEST.stub by running
  make manifeset
- Updated MANIFEST.stub with missing files
- Moved sin-oscar.spec to .in so that .spec will be generated when "make rpm"
  is run.  Version info only needs to be updated in Makefile.PL
- Added "make realclean" target to do real cleanup
- Added documentation on how to build binary packages in INSTALL
- Now generates a compatible autoinstallscript.conf such that SystemImager
  3.7.4+ will generate a .master script correctly marking the partition as
  swap (closes #260)
- New option for mksiimage "--force" which allows an image to be overwritten
  if image already exists (closes #289)
